andela/ah-scorpion
# #159053988 Add functionality for users to be able to comment on articles

**What does this PR do?**
- Allow users to make and delete comments

**Description of tasks completed**
- Create endpoints for CRUD operations of comments
- Create a  model for comments
- Make a column in the comments model for child comments(replies)
- Add unit tests for the comments functionality

**How can this PR be manually tested**
- Git pull the source code from ft-159053988-Comment-on-articles branch
- Copy the variables from `sample.env` and put the correct parameters in `.env` file
- Run `source .env` on your terminal to export environmental variables and activate virtual env
- Run `python manage.py migrate` to apply recent migrations
- Run `pip install -r requirements` to install latest dependencies
- Run `python manage.py runserver` to start the project
- If not already done, create a new user account and login
- Create an article and copy the slug
- Make a POST request to the following endpoint `api/v1/<str: slug>/comments/` replacing the `<slug>` with the slug you copied above. The data should be a comment 
`{"content": "Sample comment"}`
- You should get the comment back. 
- Make a DELETE request to the following endpoint `api/v1/<str: slug>/comments/<int: id>/` replacing the <id> with the id of the comment you got from the delete request you copied above. This returns 204_NO_CONTENT
- Make a GET request with the endpoint `api/v1/<str: slug>/comments/` and view all the comments you have made to the article from which you have taken the slug.

**Relevant Stories**
[159053988](https://www.pivotaltracker.com/n/projects/2184741/stories/159053988)
